### PR TITLE
Liberate AWSomely

### PR DESCRIPTION
**Pull request by SeamusMacRat**
- Clean up of code, including the removal of app.config files and config reader methods
- Rename of namespace to Liberator.AWSome to keep it in line with other tools
- Rename of Piping to PipeBomb, because AWSome.PipeBomb is funnier
- Rename of Messages to Chatter, because I wanted to keep the irreverent feel of the libraries

### CODE REVIEW

File: AWSome.Bucket/Config/BucketException.cs [hunk 1]
```diff
@@ -1,10 +1,18 @@
 ﻿using Amazon.S3;
 using System;
 
-namespace AWSome.Bucket.Config
+namespace Liberator.AWSome.Bucket.Config
 {
+    /// <summary>
+    ///
```
Comment (delahayeoglafa) [inline AWSome.Bucket/Config/BucketException.cs]: Summary is empty?

File: AWSome.Bucket/Config/BucketException.cs [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
-namespace AWSome.Bucket.Config
+namespace Liberator.AWSome.Bucket.Config
 {
+    /// <summary>
+    /// 
+    /// </summary>
     public class BucketException : Exception
     {
+        /// <summary>
+        ///
```
Comment (delahayeoglafa) [inline AWSome.Bucket/Config/BucketException.cs]: summay is empty?

File: AWSome.Bucket/Config/Preferences.cs [hunk 3]
```diff
@@ -2,32 +2,38 @@
...
-        // <returns>The Dynamo DB Client as configured</returns>
+        public static string ProfileLocation { get; set; }
+        
+        /// <summary>
+        /// Gets the Amazon S3 Client set in the App.config file
+        /// </summary>
+        /// <returns>The Amazon S3 Client as configured</returns>
         public static IAmazonS3 GetAmazonS3Client()
         {
             UserAWSCredentials = GetAWSCredentials();
```
Comment (delahayeoglafa) [inline AWSome.Bucket/Config/Preferences.cs:39]: Lots of empty summaries in this class? On purpose?

File: AWSome.Chatter/Objects/MessageObject.cs [hunk 4]
```diff
@@ -1,15 +1,22 @@
...
         [XmlIgnore]
         public string PartitionKey { get; set; }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="PartitionKey"></param>
         protected MessageObject(string PartitionKey)
         {
             this.PartitionKey = PartitionKey;
```
Comment (delahayeoglafa) [inline AWSome.Chatter/Objects/MessageObject.cs:22]: empty summaries in this class?

File: AWSome.Chatter/Transactions/KinesisReceiver.cs [hunk 5]
```diff
@@ -36,6 +36,7 @@ public static List<Record> GetRecordsFromKinesisQueue(DateTime fromDateTime)
         /// </summary>
         /// <param name="kinesisClient">The Kinesis client object</param>
         /// <param name="describeResponse">The stream response descriptor</param>
+        /// <param name="fromDateTime"></param>
```
Comment (delahayeoglafa) [inline AWSome.Chatter/Transactions/KinesisReceiver.cs:39]: hmmm looking back on this I think the name of fromDateTime is a little confusing - is this getting the messages after a specific DateTime? Perhaps we should rename it? What do you think?

File: AWSome.ECStraordinary/Config/Preferences.cs [hunk 6]
```diff
@@ -6,15 +6,36 @@
...
         public static RegionEndpoint RegionEndpoint { get; set; }
 
+        /// <summary>
+        /// 
+        /// </summary>
         public static string ProfileName { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
```
Comment (delahayeoglafa) [inline AWSome.ECStraordinary/Config/Preferences.cs:31]: empty summaries in this class

File: AWSome.ECStraordinary/Config/ECSException.cs [hunk 7]
```diff
@@ -2,10 +2,18 @@
...
+    /// <summary>
+    /// 
+    /// </summary>
     public class ECSException : Exception
     {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="message"></param>
+        /// <param name="exception"></param>
```
Comment (delahayeoglafa) [inline AWSome.ECStraordinary/Config/ECSException.cs]: empty summaries here

File: AWSome.ECStraordinary/Client/ECSService.cs [hunk 8]
```diff
@@ -1,21 +1,29 @@
...
     {
-
+        /// <summary>
+        /// 
+        /// </summary>
         public AWSCredentials UserAWSCredentials = Preferences.GetAWSCredentials();
 
+        /// <summary>
+        /// 
+        /// </summary>
```
Comment (delahayeoglafa) [inline AWSome.ECStraordinary/Client/ECSService.cs:26]: empty summaries

File: AWSome.Dynamite/Objects/TableObject.cs [hunk 9]
```diff
@@ -1,7 +1,10 @@
 ﻿using Amazon.DynamoDBv2.DocumentModel;
 
-namespace AWSome.Dynamite.Objects
+namespace Liberator.AWSome.Dynamite.Objects
 {
+    /// <summary>
+    /// abstract basis for a table object
+    /// </summary>
     public abstract class TableObject
     {
```
Comment (delahayeoglafa) [inline AWSome.Dynamite/Objects/TableObject.cs:9]: couple of empty summaries here

File: AWSome.Dynamite/Config/Preferences.cs [hunk 10]
```diff
@@ -56,6 +90,10 @@ public static IAmazonDynamoDB GetDynamoDbClient()
             return new AmazonDynamoDBClient(UserAWSCredentials, RegionEndpoint);
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <returns></returns>
```
Comment (delahayeoglafa) [inline AWSome.Dynamite/Config/Preferences.cs]: lots of empty summaries in this class

File: AWSome.Bucket/Config/BucketException.cs (hunk 2 above)
Comment (delahayeoglafa) [reply AWSome.Bucket/Config/BucketException.cs]: would help if I spelt it correctly - *summary

File: AWSome.Bucket/Config/BucketException.cs (hunk 1 above)
Comment (SeamusMacRat) [reply AWSome.Bucket/Config/BucketException.cs]: This exceptional bucket is no longer empty

File: AWSome.Bucket/Config/BucketException.cs (hunk 2 above)
Comment (SeamusMacRat) [reply AWSome.Bucket/Config/BucketException.cs]: This bucket is definitely exceptional

File: AWSome.Bucket/Config/Preferences.cs (hunk 3 above)
Comment (SeamusMacRat) [reply AWSome.Bucket/Config/Preferences.cs:39]: Yes …

… nope, can't keep up the lie. I thought I had filled them all in. LOL

File: AWSome.Chatter/Objects/MessageObject.cs (hunk 4 above)
Comment (SeamusMacRat) [reply AWSome.Chatter/Objects/MessageObject.cs:22]: How did that happen? Oh, yeah ... me ... d'oh!

File: AWSome.Chatter/Transactions/KinesisReceiver.cs (hunk 5 above)
Comment (SeamusMacRat) [reply AWSome.Chatter/Transactions/KinesisReceiver.cs:39]: It looks to be the timestamp for the first record to fetch an iterator for.

Does timestamp seem like the better name? Or firstTimestamp? Will leave as is for now until you let me know what you think :)

File: AWSome.ECStraordinary/Config/Preferences.cs (hunk 6 above)
Comment (SeamusMacRat) [reply AWSome.ECStraordinary/Config/Preferences.cs:31]: Sorted :)

File: AWSome.ECStraordinary/Config/ECSException.cs (hunk 7 above)
Comment (SeamusMacRat) [reply AWSome.ECStraordinary/Config/ECSException.cs]: Done

File: AWSome.ECStraordinary/Client/ECSService.cs (hunk 8 above)
Comment (SeamusMacRat) [reply AWSome.ECStraordinary/Client/ECSService.cs:26]: 💃 Done

File: AWSome.Dynamite/Objects/TableObject.cs (hunk 9 above)
Comment (SeamusMacRat) [reply AWSome.Dynamite/Objects/TableObject.cs:9]: Not any more ... he says hopefully

File: AWSome.Dynamite/Config/Preferences.cs (hunk 10 above)
Comment (SeamusMacRat) [reply AWSome.Dynamite/Config/Preferences.cs]: Naaah ... I reckon there aren't now :)

**Review by delahayeoglafa — APPROVED** [review]
LGTM

File: AWSome.Dynamite/Client/DynamoService.cs [hunk 11]
```diff
@@ -156,6 +160,7 @@ private ScanFilter AddScanFilterParameter(ScanFilter scanFilter, string attribut
         /// </summary>
         /// <param name="tableName">Name of table to delete items from</param>
         /// <param name="keyName">Name of key which will be used to delete all items with it</param>
+        /// <param name="sortkey"></param>
```
Comment (delahayeoglafa) [inline AWSome.Dynamite/Client/DynamoService.cs:163]: uh oh...another empty param

**Pull request merged**